### PR TITLE
Shuna146

### PR DESCRIPTION
**Pull request by JoannaCode**
@FreeTymeKiyan

### CODE REVIEW

File: Hard/LRUCache_shuna.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    private HashMap<Integer, DoubleLinkedList> map = new HashMap<Integer, DoubleLinkedList>();
+    private int capacity;
+    private int count;
+    private DoubleLinkedList head;
+    private DoubleLinkedList tail;
+    
+    public LRUCache(int capacity) {
+        this.capacity = capacity;
+        count = 0;
+        map = new HashMap<>();
```
Comment (FreeTymeKiyan) [inline Hard/LRUCache_shuna.java]: you initialized map already at line 24, why again?

File: Hard/LRUCache_shuna.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    
+    public void addToHead(DoubleLinkedList node) {
+        node.next = head.next;
+        node.next.pre = node;
+        node.pre = head;
+        head.next = node;
+    }
+    
+    public int get(int key) {
+        if(map.get(key) != null) {
```
Comment (FreeTymeKiyan) [inline Hard/LRUCache_shuna.java]: there is a `map.containsKey` method to check existence

**Comment by JoannaCode** [comment]
@FreeTymeKiyan  fixed!

**Comment by FreeTymeKiyan** [comment]
LGTM

**Pull request merged**